### PR TITLE
The profiler now starts earlier into the init process. + Documents the initialization order and pipeline

### PR DESCRIPTION
**Pull request by MrsTonedOne**
While working over in my head what the init order was (again) I decided it was time to properly put it on paper.

### CODE REVIEW

File: code/world.dm [hunk 1]
```diff
@@ -22,3 +22,8 @@
 #ifdef FIND_REF_NO_CHECK_TICK
 	loop_checks = FALSE
 #endif
+
+#if DM_BUILD >= 1506
+// We don't actually care about storing the output here, this is just an easy way to ensure the profile runs first.
+var/world_init_profiler = world.Profile(PROFILE_START)
+#endif
```
Comment (Rohesie) [inline code/world.dm:29]:
```suggestion
#endif

```
Linter appeasement.

File: code/world.dm [hunk 2]
```diff
@@ -22,3 +22,8 @@
 #ifdef FIND_REF_NO_CHECK_TICK
 	loop_checks = FALSE
 #endif
+
+#if DM_BUILD >= 1506
+// We don't actually care about storing the output here, this is just an easy way to ensure the profile runs first.
+GLOBAL_REAL_VAR(world_init_profiler) = world.Profile(PROFILE_START)
```
Comment (Rohesie) [inline code/world.dm:28]: This is not a static value, and as such won't compile.

Comment (MrStonedOne) [reply code/world.dm:28]: not the issue. try again

File: code/world.dm (hunk 1 above)
Comment (MrStonedOne) [reply code/world.dm]: the linter was mainly mad about the lack of use of global helpers.

File: code/world.dm (hunk 2 above)
Comment (Rohesie) [reply code/world.dm:28]: Looks like `world.dm` is read before `\_DEFINES\global.dm`, so the macro is not yet defined when this code is executed.

Comment (Cyberboss) [reply code/world.dm:28]: I think I'd put this in `__DEFINES/_profiler_setup.dm`

Comment (MrStonedOne) [reply code/world.dm:28]: I was thinking about just doing `var world_init_profile = ...` =P

Comment (Cyberboss) [reply code/world.dm:28]: Doesn't matter too much really. Someone is aiming to improve file layout/organization soon, so we'll see how that goes.

File: code/__DEFINES/_profile.dm [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#if DM_BUILD >= 1506
+// We don't actually care about storing the output here, this is just an easy way to ensure the profile runs first.
+GLOBAL_REAL_VAR(world_init_profiler) = world.Profile(PROFILE_START)
+#endif
```
Comment (Rohesie) [inline code/__DEFINES/_profile.dm:4]:
```suggestion
#endif

```
Linter no like

Comment (MrStonedOne) [reply code/__DEFINES/_profile.dm:4]: I did that already what the fuck!

Comment (MrStonedOne) [reply code/__DEFINES/_profile.dm:4]: oh, helps if i push.

**Pull request merged**